### ISSUE TITLE
Add measurement errors in mock data

### ISSUE THREAD

**Issue by combet** [LSSTDESC/CLMM#394]
At the moment, mock data does not allow to generate measurement errors on the observed ellipticities, contrarily to what shape measurements codes produce. This should be implemented.

**Comment by cristobal-sifon** [comment]
If we add this we should also probably add functionality to deal with it in ``clmm``. What would be the plan for this? And also, I worry that we would need to consider correlation in uncertaintes between redshifts and ellipticities, unless we make it very clear that the resulting catalog cannot be subsampled in any way (save randomly?)

**Comment by combet** [comment]
@cristobal-sifon, yes of course of need to deal with it in `clmm`. So in my mind, the first place where those errors would be used would be in weights to be added to the evaluation of the shear profile (g_t or DeltaSigma), which is somewhat contained in #395 for which @m-aguena already opened a PR. It is also linked to #237 (which was open a while ago). Providing the actual weights for the DeltaSigma profile, which use the error on the shape but not only (see eqs in #396), has not yet been flag as an issue, but should be (feel free to open an new issue). Am I forgetting other places where these errors would come into play?

Regarding your second point, could you clarify please? do you mean that there is a correlation as more distant galaxies will generally have large ellipticity uncertainties?

**Comment by cristobal-sifon** [comment]
Thanks @combet 

> Regarding your second point, could you clarify please? do you mean that there is a correlation as more distant galaxies will generally have large ellipticity uncertainties?

yes that's what I mean, broadly speaking. More specifically both the redshift and its uncertainty, as well as the uncertainty on the ellipticity, depend on the brightness and size of the galaxy - and survey depth, of course. This is not a simple issue, and we should consider leaving it out of `clmm`. However if we really want to have this I would suggest reaching out to HSC to see if we can get a realistic joint distribution of all these quantities (not the actual catalog but an n-dimensional histogram) from which to draw them for our mock samples. Would be good to discuss

**Comment by combet** [comment]
Thanks for the clarification @cristobal-sifon. I agree that this is a complex problem. My initial thought was to generate shape errors independently of redshifts in `mock_data`. That would not be realistic, but would allow us to exercise the other items/steps of `clmm` that do need to account for those errors. In parallel, we could indeed try to get the information we'd need to make this realistic, but this could be seen as a second step. Let's discuss this is more details at a future tag-up/co-work session.

Also pinging @shenmingfu who first suggested that addition to `mock_data`.

**Comment by cristobal-sifon** [comment]
Oh we could have a private method to test the implementation, that's fine, but anything we want to make part of what the user sees we should be very careful

**Comment by cristobal-sifon** [comment]
After discussion, we've agreed to add a column to `galcat` with shape measurement error while in parallel implementing a way to account for them (cf. #395,#396). We will make this column visible but we need a big "WARNING: EXPERIMENTAL" sign somewhere (in the docs for sure - do we want to print one when producing the mock catalog?). Until we can build an error distribution that accounts for the correlation with redshift (and this could come from e.g. HSC o DC2 mocks), the mock source sample cannot be subsampled while still using the uncertainties.

**Comment by marina-ricci** [comment]
@combet @cristobal-sifon what is the status of that issue?
